ts/main.ts

```diff
@@ -1 +1,55 @@
 /* global data */
+
+interface FormElements extends HTMLFormControlsCollection {
+  photoUrl: HTMLInputElement;
+  title: HTMLInputElement;
+  notes: HTMLTextAreaElement;
+}
+
+const $image = document.getElementsByTagName('img');
+const imf = $image[0] as HTMLImageElement;
+const $newImg = document.getElementById('photo-url') as HTMLInputElement;
+const $form = document.querySelector('#form') as HTMLFormElement;
+
+function imgChanger(event: Event): void {
+  const $change = event.target as HTMLInputElement;
+  const picture = $change.value;
+  if (urlChecker(picture) === true) {
+    imf.src = picture;
+  } else {
+    imf.src =
+      'https://www.pokemon.com/static-assets/content-assets/cms2/img/pokedex/full/403.png';
+  }
+}
+
+function urlChecker(string: string): boolean {
+  const res = string.match(
+    /(http(s)?:\/\/.)?(www\.)?[-a-zA-Z0-9@:%._+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:%_+.~#?&//=]*)/g,
+  );
+  return res !== null;
+}
+
+if ($newImg) $newImg.addEventListener('change', imgChanger);
+if (!$newImg) throw new Error('The Image did not load.');
+
```

Comment:
You don't need both of these. You should move your line 33 check above line 32 and then remove your `if ($newImg)` from line 32 because it won't be necessary after you validate that your `$newImg` is not falsy